### PR TITLE
Add editor to make an existing project a basic generator

### PR DESCRIPTION
**Pull request by russmiles**
Implementation of editor that will turn an existing project into a generator.

To try it out create any working project and run the `ConvertExistingProjectToGenerator` editor against it.

### CODE REVIEW

**Comment by Lawouach** [comment]
LGTM. Quite useful! 

Just a couple of comments though:

* could you squash your commits to make this PR easier to read?
* I realise we rarely comment our TS code and I think we should start, what do you think? :)

File: .atomist/editors/ProjectToGeneratorOperations.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    let testFile: File = eng.scalar(project, testPathExpression);
+    testFile.replace(originalTestAssertionsString, testAssertionsString)
+}
+
+const pathToAtomist = ".atomist";
+const pathToNodeModules = "node_modules";
+const pathToTarget = "target";
+const pathToIdeaIDEArtifacts = ".idea";
+
+function notACommonPerpheralArtifact(file: File): boolean {
```
Comment (ddgenome) [inline .atomist/editors/ProjectToGeneratorOperations.ts]: Perpheral > Peripheral

File: .atomist/editors/ProjectToGeneratorOperations.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { PathExpression, PathExpressionEngine } from '@atomist/rug/tree/PathExpression'
+
+export function addAssertionsForAllFilesInProject(project: Project, generatorName: string): void {
+    let originalTestAssertionsString = "then\n  fileExists \"README.md\""
+
+    let testAssertionsString = project.files().reduce(function(acc, file) {
+      if (notACommonPerpheralArtifact(file))  {
+          if(acc === "then"){
+            acc +=  "\n  "
+          }else{
```
Comment (ddgenome) [inline .atomist/editors/ProjectToGeneratorOperations.ts]: Cozy

File: .atomist/editors/ProjectToGeneratorOperations.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            acc +=  "\n    and "
+          }
+          acc += "fileExists \"" + file.path() + "\"";
```
Comment (ddgenome) [inline .atomist/editors/ProjectToGeneratorOperations.ts]: Could use single quotes or a template string to simplify here.

File: .atomist/editors/ProjectToGeneratorOperations.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    let eng: PathExpressionEngine = project.context().pathExpressionEngine();
+    let testPathExpression = new PathExpression<Project, File>("/*[@name='.atomist']/tests/*[@name='" + generatorName + ".rt']");
+    let testFile: File = eng.scalar(project, testPathExpression);
+    testFile.replace(originalTestAssertionsString, testAssertionsString)
+}
+
+const pathToAtomist = ".atomist";
+const pathToNodeModules = "node_modules";
+const pathToTarget = "target";
+const pathToIdeaIDEArtifacts = ".idea";
```
Comment (ddgenome) [inline .atomist/editors/ProjectToGeneratorOperations.ts]: Why define these constants here? The only one that is used is `pathToAtomist` and it is only used in `notACommonPeripheralArtifact`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 [Unreleased]: https://github.com/atomist-rugs/rug-editors/compare/0.11.0...HEAD
 
+- Added `ConvertExistingProjectToGenerator` editor.
```
Comment (ddgenome) [inline CHANGELOG.md]: Technically, the text of a markdown list item should start in column four.

File: CHANGELOG.md [hunk 6]
```diff
@@ -9,6 +9,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 [Unreleased]: https://github.com/atomist-rugs/rug-editors/compare/0.11.0...HEAD
```
Comment (ddgenome) [inline CHANGELOG.md]: Are you planning to cut a release after this is merged? Might be good to prep the CHANGELOG for that.

File: CHANGELOG.md (hunk 5 above)
Comment (russmiles) [inline CHANGELOG.md]: All changes should now be in, and assuming it's all good to go then this will be a squash and merge if you don't mind @ddgenome ?

**Comment by russmiles** [comment]
When it comes to comments, I'd prefer the code to be as clean as possible @Lawouach so I'm not sure what further comments would add here?

**Comment by Lawouach** [comment]
The code rarely explains the why of any function, only the how. I think that's what I had in mind. But that's a general comment not a specific one here.

**Comment by ddgenome** [comment]
Personally, I'm not a fan of redundant comments. We would probably benefit from starting to use TypeDoc for our classes, modules, functions, etc.

**Comment by Lawouach** [comment]
I have a lot of joy with code that explains **why** it exists in the first place. It doesn't mean each editor should have such section. That's fair.

**Comment by ddgenome** [comment]
I'm cleaning this up now.

**Pull request merged**